File: QE-selenium python/Activity14.py
from selenium import webdriver
from selenium.webdriver.common.by import By
import time

driver = webdriver.Chrome()
driver.get("https://training-support.net/webelements/tables")

print(driver.title)

rows = driver.find_elements(By.XPATH, "//table/tbody/tr")
cols = driver.find_elements(By.XPATH, "//table/tbody/tr[1]/td")

print("Rows:", len(rows))
print("Columns:", len(cols))

book_before = driver.find_element(By.XPATH, "//table/tbody/tr[5]/td[1]")
print("Book in 5th row before sort:", book_before.text)

driver.find_element(By.XPATH, "//table/thead/tr/th[3]").click()
time.sleep(2)

book_after = driver.find_element(By.XPATH, "//table/tbody/tr[5]/td[1]")
print("Book in 5th row after sort:", book_after.text)

driver.quit()

File: QE selenium java/Activity14.java
package demos;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import java.util.List;

public class Activity14 {
    public static void main(String[] args) throws InterruptedException {
        WebDriver driver = new ChromeDriver();
        driver.get("https://training-support.net/webelements/tables");

        System.out.println(driver.getTitle());

        List<WebElement> rows = driver.findElements(By.xpath("//table/tbody/tr"));
        List<WebElement> cols = driver.findElements(By.xpath("//table/tbody/tr[1]/td"));

        System.out.println("Rows: " + rows.size());
        System.out.println("Columns: " + cols.size());

        WebElement bookBefore = driver.findElement(By.xpath("//table/tbody/tr[5]/td[1]"));
        System.out.println("Book in 5th row before sort: " + bookBefore.getText());

        driver.findElement(By.xpath("//table/thead/tr/th[3]")).click();
        Thread.sleep(2000);

        WebElement bookAfter = driver.findElement(By.xpath("//table/tbody/tr[5]/td[1]"));
        System.out.println("Book in 5th row after sort: " + bookAfter.getText());

        driver.quit();
    }
}
